applications/dolphin/helpers/dolphin_state.c

```diff
@@ -28,148 +21,42 @@ typedef struct {
-    }
-
-    storage_file_close(file);
-    storage_file_free(file);
-
-    dolphin_state->dirty = !save_result;
-
-    FURI_LOG_I("dolphin-state", "Saved");
-
-    return save_result;
+    return saved_struct_save(
+        DOLPHIN_STORE_PATH,
+        &dolphin_state->data,
+        sizeof(DolphinStoreData),
+        DOLPHIN_STORE_HEADER_MAGIC,
+        DOLPHIN_STORE_HEADER_VERSION);
 }
 
 bool dolphin_state_load(DolphinState* dolphin_state) {
-    DolphinStore store;
-    // Read Dolphin State Store
-    FURI_LOG_I("dolphin-state", "Loading state from \"%s\"", DOLPHIN_STORE_KEY);
-
-    File* file = storage_file_alloc(dolphin_state->fs_api);
-    bool load_result = storage_file_open(file, DOLPHIN_STORE_KEY, FSAM_READ, FSOM_OPEN_EXISTING);
-    if(!load_result) {
-        FURI_LOG_E(
-            "dolphin-state",
-            "Load failed. Storage returned: %s",
-            storage_file_get_error_desc(file));
-    } else {
-        uint16_t bytes_count = storage_file_read(file, &store, sizeof(DolphinStore));
-
-        if(bytes_count != sizeof(DolphinStore)) {
-            load_result = false;
-        }
-    }
-
-    if(!load_result) {
-        FURI_LOG_E("dolphin-state", "DolphinStore size mismatch");
-    } else {
-        if(store.header.magic == DOLPHIN_STORE_HEADER_MAGIC &&
-           store.header.version == DOLPHIN_STORE_HEADER_VERSION) {
-            FURI_LOG_I(
-                "dolphin-state",
-                "Magic(%d) and Version(%d) match",
-                store.header.magic,
-                store.header.version);
-            uint8_t checksum = 0;
-            const uint8_t* source = (const uint8_t*)&store.data;
-            for(size_t i = 0; i < sizeof(DolphinStoreData); i++) {
-                checksum += source[i];
-            }
-
-            if(store.header.checksum == checksum) {
-                FURI_LOG_I("dolphin-state", "Checksum(%d) match", store.header.checksum);
-                dolphin_state->data = store.data;
-            } else {
-                FURI_LOG_E(
-                    "dolphin-state",
-                    "Checksum(%d != %d) mismatch",
-                    store.header.checksum,
-                    checksum);
-                load_result = false;
-            }
-        } else {
-            FURI_LOG_E(
-                "dolphin-state",
-                "Magic(%d != %d) or Version(%d != %d) mismatch",
-                store.header.magic,
-                DOLPHIN_STORE_HEADER_MAGIC,
-                store.header.version,
-                DOLPHIN_STORE_HEADER_VERSION);
-            load_result = false;
-        }
+    bool loaded = saved_struct_load(
+        DOLPHIN_STORE_PATH,
+        &dolphin_state->data,
+        sizeof(DolphinStoreData),
+        DOLPHIN_STORE_HEADER_MAGIC,
+        DOLPHIN_STORE_HEADER_VERSION);
+    if(!loaded) {
+        FURI_LOG_W("dolphin-state", "Reset dolphin-state");
+        memset(dolphin_state, 0, sizeof(*dolphin_state));
```

Comment:
Why not sizeof(DolphinState)

Reply:
It seems safer. If you change structure type you automatically get correct size here.